tests/lib/tools/nested-state

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: prepare"
+    echo "       restore"
+    echo "       build-image IMAGE-TYPE"
+    echo "       boot-image IMAGE-TYPE [--param-cd PARAM] [--param-mem PARAM]"
+    echo "       start-vm"
+    echo "       shutdown-vm"
+    echo "       delete-vm"
+    echo ""
+    echo "Available options:"
+    echo "  -h --help   show this help message."
+    echo ""
+    echo "COMMANDS:"
+    echo "  prepare:     creates all the directories needed to run a nested test"
+    echo "  restore:     deletes all the directories and data used by nested tests"
+    echo "  build-image: creates an image using ubuntu image tool"
+    echo "  boot-image:  first boot of the image which it remains running"
+    echo "  start-vm:    start the booted vm"
```

Comment:
This part is more problematic, by definition a booted system is running. Does it mean that it can start a virtual machine that was previosuly created with "build-image" or that it must have been booted before with "boot-image"?

This is also confusing because "boot-image" doesn't read right - you cannot boot an image, you can boot a system and that's just what "start-vm" is. I'm surely missing the critical distinction between the two. Could you explain this please?

Reply:
Thanks for the suggestion:

So now the commands are:

prepare:     creates all the directories needed to run a nested test
restore:     removes all the directories and data used by nested tests
build-image: creates an image using ubuntu image tool
create-vm:   creates a vm and run the first boot
start-vm:    starts a stopped vm
stop-vm:     shutdowns a running vm
remove-vm:   removes a vm

Reply:
This is great, thank you!